nix/build-dapp-package.nix

```diff
@@ -0,0 +1,88 @@
+{ pkgs }:
+
+let
+  remappings = xs:
+    builtins.foldl' pkgs.lib.mergeAttrs {}
+      (builtins.map
+        (x: {
+          "${x.name}/" = "${x}/dapp/${x.name}/src/";
+         } // x.remappings)
+         xs);
+  libPaths = xs:
+    builtins.foldl' pkgs.lib.mergeAttrs {}
+      (builtins.map
+        (x: {
+          "${x.name}" = "${x}/dapp/${x.name}/src";
+         } // x.libPaths)
+         xs);
+in
+  {
+    name
+  , src
+  , deps ? []
+  , solc ? pkgs.solc
+  , test-hevm ? pkgs.dapp2.test-hevm
+  , solcFlags ? ""
+  , doCheck ? true
+  , extract ? false
+  , ... }@args:
+    pkgs.stdenv.mkDerivation ( rec {
+      inherit doCheck extract;
+
+      buildInputs = [ pkgs.jq solc test-hevm ];
+      passthru = {
+        remappings = remappings deps;
+        libPaths = libPaths deps;
+      };
+
+      REMAPPINGS =
+        pkgs.lib.mapAttrsToList
+          (k: v: k + "=" + v)
+          passthru.remappings;
+
+      LIBSCRIPT =
+        pkgs.lib.mapAttrsToList
+          (k: v: ''
+            ln -s ${v} lib/${k}
+          '')
+          passthru.libPaths;
+
+      buildPhase = args.buildPhase or ''
```

Comment:
Is there any way we could merge this and `dapp build` somehow? This script is essentially `dapp build`. It seems that whenever we change `dapp build` we'd also need to change it here.

Reply:
Hmm yeah, that's a good point.

Maybe we could extract the bash stuff in the different phases into their own little script, and use those both in `build-dapp-package` and in `dapp-build`.

Is it OK if I file an issue for this or do you think it should be in this PR?

Reply:
Also honestly, inlined Bash inside of Nix is pretty gross. It would be good to move it out for that reason too.

Reply:
Yeah, we can def postpone this for a later pr